src/gz/watchlist.c

```diff
@@ -191,6 +192,12 @@ static int add_member(struct item_data *data,
   member_data->position_set = 1;
   menu_add_button_icon(imenu, 0, 0, list_icons, 1, 0xFF0000,
                        remove_button_proc, member_data);
+
+  if (!menu_checkbox_get(data->visibility_checkbox)) {
```

Comment:
This fixes the bug where the checkbox state wasn't being respected when creating new watches.